docs/rfc/rfc-1.adoc

```diff
@@ -633,13 +633,148 @@ As time passes and operators drop out of the system, a wallet becomes at risk
-a <<heartbeat,heartbeat>>), we motion to transfer the funds to another randomly
-selected wallet.
+than the `wallet_max_age`, or if it drops below the liveness threshold (say,
+below 70 on a <<heartbeat,heartbeat>>), we motion to
+<<closing-a-wallet,transfer the funds>> to another randomly selected wallet.
 
 Once a wallet no longer has funds and is not the primary wallet for new
-deposits, it can be closed and operators are no longer required to maintain
-it.
+deposits, it can be <<closing-a-wallet,closed>> and operators are no longer
+required to maintain it.
+
+[[closing-a-wallet]]
+==== Closing A Wallet
+
+Governable Parameters:
+
+- `heartbeat`: The number of group members required for a
+  <<heartbeat,heartbeat>> to be successful.
+- `wallet_closure_timeout`: The amount of time that a wallet has to
+  successfully move its funds to another wallet and inform the ethereum chain
+  before it is at risk of punishment.
+- `wallet_max_age`: The oldest we allow a wallet to become before we transfer the funds
+  to a randomly selected wallet.
+- `wallet_min_btc`: The smallest amount of btc a wallet can hold before we
+  attempt to close the wallet and transfer the funds to a randomly selected
+  wallet.
+
+When a wallet fails a <<heartbeat,heartbeat>> `consecutive_failed_heartbeats`
+times, ends a <<redemption,redemption>> with less than `wallet_min_btc`
+remaining, or exceeds the `wallet_max_age`, the operators begin the process of
+closing it. If there are still funds remaining, the <<operator-only,first
+operator>> posts a <<operator-only,reimbursable>> transaction to the ethereum
+chain declaring an intention to close the wallet. This transaction includes a
+timestamp hereby referred to as `move_funds_start_timestamp`.
+
+That operator proposes a BTC fee (as in <<sweeping,sweeping>>) and the
+operators use the wallet's public key hash to choose viable (it must also have
+a passing heartbeat and be above `wallet_min_btc`, and not exceeded the
+`wallet_max_age) wallet(s) to transfer the remaining funds to. They construct a
+P2PKH transaction moving the wallet's main UTXO to that wallet. For more
+details on exactly how a wallet is chosen and how the funds are split up see
+<<wallet-closure-transaction,the dedicated section>>.
+
+After this transaction is complete, <<public-knowledge,the public>> is able to
+submit a SPV proof to let ethereum know that the funds were transferred
+nonfraudulently. Later, a <<donate,donation>> can be made by governance to
+handle any difference in outstanding v2 token supply and locked BTC due to
+mining transfer fees.
+
+Later, <<public-knowledge,the public>> at any time can challenge the operator's
+wallet choice. They do so by showing that there existed a live, valid wallet
+which was created after `move_funds_start_timestamp` that _wasn't_ chosen _and_
+should have been. These challenges are incentivized to happen sooner rather
```

Comment:
```suggestion
Later, <<public-knowledge,the public>> at any time can challenge the operator's
wallet choice. They do so by showing that there exists a live, valid wallet
which was created before `move_funds_start_timestamp` that _wasn't_ chosen _and_
should have been. These challenges are incentivized to happen sooner rather
```

We can not prove on-chain there was a live wallet without additional costs of tracking timestamps of wallet status updates. But I think it is a typo given the created before/after part. What we can prove easily in the challenge: there is a live wallet, created before `move_funds_start_timestamp` and that wallet should be chosen but it wasn't.

Reply:
yeah - definitely meant before! [dd68fe7](https://github.com/keep-network/tbtc-v2/pull/127/commits/dd68fe7bb46201b906f6d41f3cbcca28fb9e907b)